build_runner/lib/src/watcher/node_watcher.dart

```diff
@@ -40,9 +44,12 @@ class PackageNodeWatcher {
   /// packageA.watch()
   /// ```
   Stream<AssetChange> watch([String relative]) {
```

Comment:
Given that this is now stateful and has new restrictions on how it can be called I wonder if we should refactor a bit. I don't have any concrete suggestions so feel free to to defer and I can take a deeper look later.